File: LeetCode Questions + Tutorials/EngineeringDigest/Leetcode1512.java
import java.util.*;

class Leetcode1512 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the size of the array: ");
        int size = input.nextInt();
        int[] arr = new int[size];
        for(int i = 0; i < size; i++) {
            System.out.printf("Enter %d number element: ", i+1);
            arr[i] = input.nextInt();
        }
        input.close();
        System.out.println(numIdenticalPairs(arr));
    }
    private static int numIdenticalPairs(int[] nums) {
        int sum = 0;
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i : nums) {
            map.put(i, map.getOrDefault(i, 0) + 1);
        }
        for(Map.Entry<Integer, Integer> entry : map.entrySet()) {
            sum += (entry.getValue() * (entry.getValue() - 1)) / 2;
        }
        return sum;
    }
}

File: LeetCode Questions + Tutorials/EngineeringDigest/Leetcode1512.js
function numIdenticalPairs(nums) {
    var map = new Map();
    var sum = 0;
    for (var _i = 0, nums_1 = nums; _i < nums_1.length; _i++) {
        var i = nums_1[_i];
        map.set(i, (map.get(i) || 0) + 1);
    }
    console.log('Map contents:', map);
    console.log('Starting loop');
    for (var _a = 0, _b = map.entries(); _a < _b.length; _a++) {
        var _c = _b[_a], key = _c[0], value = _c[1];
        console.log("Value: ".concat(value, ", Key: ").concat(key));
        sum += (value * (value - 1)) / 2;
    }
    return sum;
}
;
var nums = [1, 2, 3, 1, 1, 3];
console.log(numIdenticalPairs(nums));
